flexmeasures/data/models/planning/storage.py

```diff
@@ -833,25 +900,37 @@ def compute(self, skip_validation: bool = False) -> SchedulerOutputType:
         ) = self._prepare(skip_validation=skip_validation)
 
         # Fallback policy if the problem was unsolvable
-        storage_schedule = fallback_charging_policy(
-            sensor, device_constraints[0], start, end, resolution
-        )
-        storage_schedule = convert_units(storage_schedule, "MW", sensor.unit)
+        storage_schedule = {
+            sensor: fallback_charging_policy(
+                sensor, device_constraints[0], start, end, resolution
```

Comment:
This is still wrong (and missed by tests). The fallback schedule for device `d` should be based on device constraints for device `d` rather than device `0`.